gui/notes/index.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @module notes
+ * @flow
+ */
+
+const { models } = require('cozy-client')
+
+const path = require('path')
+const metadata = require('../../core/metadata')
+const {
+  CozyDocumentMissingError,
+  CozyMissingPermissionsError,
+  UnreachableError
+} = require('../../core/remote/errors')
+
+const logger = require('../../core/utils/logger')
+const log = logger({
+  component: 'Notes'
+})
+
+/*::
+import { Shell } from 'electron'
+import { App } from '../../core/app'
+import { Config } from '../../core/config'
+import { Pouch } from '../../core/pouch'
+import { Remote } from '../../core/remote'
+import type { Metadata } from '../../core/metadata'
+
+type CozyFile = Object
+*/
+
+/**
+ * computeNoteURL - Constructs a link to the Note within the web app
+ *
+ * @param {object} Options               Object of options
+ * @param {string}   options.noteId        Id of the CouchDB document representing the given note
+ * @param {string}   options.cozyUrl       Base URL of the cozy, eg. cozy.tools or test.mycozy.cloud
+ * @param {string}   options.subDomainType Whether the cozy is using flat or nested subdomains. Defaults to flat.
+ *
+ * @return {string} Generated URL
+ */
+const computeNoteURL = async ({ noteId, cozyUrl, subDomainType }) => {
```

Comment:
I thought that this function would be shared between apps. Is there a reason to not add it to cozy-client?

Reply:
@nono +1.

We already have something similar : https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/note.js#L6

Reply:
Ah yes, I added this function at the very beginning of my experiments when I was not using `cozy-client`.
I'll take a look at how I can replace that.